.github/workflows/build.yml

```diff
@@ -29,7 +29,28 @@ jobs:
       fail-fast: false
     steps:
       - uses: actions/checkout@v2
+      # This speeds up the publish-test script, which would otherwise need to
+      # build tuftool and pubsys every time, saving several minutes.  We don't
+      # cache CARGO_HOME because that's shared with the OS build.
+      - name: Cache Bottlerocket tools build artifacts
+        uses: actions/cache@v2
+        with:
+          path: |
+            tools/bin
+            tools/.crates.toml
+            tools/.crates2.json
+            tools/target
+          # You can edit the .github/tools_cache_bust file if you need to clear the cache
+          key: ${{ hashFiles('.github/tools_cache_bust') }}-${{ runner.os }}-${{ hashFiles('tools/Cargo.lock') }}
+          restore-keys: |
+            ${{ hashFiles('.github/tools_cache_bust') }}-${{ runner.os }}-
       - run: rustup toolchain install 1.53.0 && rustup default 1.53.0
       - run: cargo install --version 0.30.0 cargo-make
+      - run: cargo install --version 0.6.2 cargo-sweep
+      - run: |
+          cargo sweep -i -r tools/
```

Comment:
If we have PRs out that change the build host's rust version, will this cause us to thrash when the workflow runs with and without the change?

Reply:
The cache, as it's currently implemented, will not be shared across PRs so there should be no interference. 

If we end up with a shared cache from the base branch, there should still be no interference since that shared cache from the develop branch shouldn't be overwritten by a PR's workflow.